.github/workflows/pr_build.yml

```diff
@@ -40,12 +40,31 @@ env:
   RUST_VERSION: stable
 
 jobs:
+  linux-test-rust:
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        is_push_event:
+          - ${{ github.event_name == 'push' }}
+      fail-fast: false
+    name: ${{ matrix.os }}/rust
+    runs-on: ${{ matrix.os }}
+    container:
+      image: amd64/rust
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Rust & Java toolchain
+        uses: ./.github/actions/setup-builder
+        with:
+          rust-version: ${{env.RUST_VERSION}}
+      - name: Rust test steps
+        uses: ./.github/actions/rust-test
```

Comment:
What about `upload-test-reports`?

Reply:
`upload-test-reports` is a parameter to Java tests (`./.github/actions/java-test`) do id not needed for the Rust jobs.